dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -2044,7 +2059,7 @@ private String getOrgUnitSql( SqlHelper hlp, EventSearchParams params, List<Orga
 
         if ( params.getOrgUnit() != null && !params.isPathOrganisationUnitMode() )
         {
-            orgUnitSql.append( " ou.organisationunitid = " )
+            orgUnitSql.append( " psi.organisationunitid = " )
```

Comment:
Immediate Reaction:  I think this change should also be conditional. Since the joining condition either uses the ownership org unit or the psi org unit. For Tracker programs (WITH_REGISTRATION), this clause shall remain as `ou.organisationunitid`. For Event Programs (WITHOUT_REGISTRATION), since ownership is completely out of scope it is okay to change to `psi.organisationunitid`. 


Reaction after thinking for a bit: Does this specific line change really introduce a performance benefit? Even though HISP Bangladesh states that this change in query seems to improve performance, it might be worth confirming with actual stress test numbers.

Reply:
On lines 1070 - 1087, we already "link" ou.organisationunitid to psi or po (Depending on condition). This line should in theory not actually make a difference, the set should already be the same :)

Worst case, without a condition for WITH/WITOUT registration, this will return the wrong result (Since po.organisationunitid != psi.organisationunitid; We would filter on the wrong thing).